packages/ember-htmlbars/lib/component.js

```diff
@@ -138,6 +139,21 @@ const Component = View.extend(TargetActionSupport, {
     }
   }),
 
+  templateForName(name, type) {
```

Comment:
It isn't just RE: outlet anyways. This method is how we support layoutName (in component's init). I'm happy to refactor in the future, but IMO this is definitely better already...